internal/commands/watch.go

```diff
@@ -0,0 +1,69 @@
+package commands
+
+import (
+	"time"
+
+	"github.com/authzed/zed/internal/client"
+	"github.com/authzed/zed/internal/console"
+
+	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
+	"github.com/spf13/cobra"
+)
+
+var (
+	watchObjectTypes []string
+	watchRevision    string
+)
+
+func RegisterWatchCmd(rootCmd *cobra.Command) *cobra.Command {
+	rootCmd.AddCommand(watchCmd)
+
+	watchCmd.Flags().StringSliceVar(&watchObjectTypes, "object_types", nil, "optional revision at which to check")
+	watchCmd.Flags().StringVar(&watchRevision, "revision", "", "optional revision at which to check")
+	return watchCmd
+}
+
+var watchCmd = &cobra.Command{
+	Use:   "watch [object_types, ...] [start_cursor]",
+	Short: "watches the stream of relationship updates from the server",
+	Args:  cobra.RangeArgs(0, 2),
+	RunE:  watchCmdFunc,
+}
+
+func watchCmdFunc(cmd *cobra.Command, _ []string) error {
+	console.Printf("starting watch stream over types %v and revision %v\n", watchObjectTypes, watchRevision)
+
+	cli, err := client.NewClient(cmd)
+	if err != nil {
+		return err
+	}
+
+	req := &v1.WatchRequest{
+		OptionalObjectTypes: watchObjectTypes,
+	}
+	if watchRevision != "" {
+		req.OptionalStartCursor = &v1.ZedToken{Token: watchRevision}
+	}
+
+	watchStream, err := cli.Watch(cmd.Context(), req)
+	if err != nil {
+		return err
+	}
+
+	for {
+		select {
+		case <-cmd.Context().Done():
+			console.Println("context done, terminating stream")
+			return nil
+		default:
+			resp, err := watchStream.Recv()
+			if err != nil {
+				return err
+			}
+
+			for _, update := range resp.Updates {
+				console.Printf("%v: %v\n", time.Now(), update)
```

Comment:
maybe the command should take a `--timestamp` flag for whether you want to see timestamps?